### PR TITLE
{CI} Breaking change tool integration

### PR DESCRIPTION
**Pull request by wangzelin007**
---
Support comparing the current PR with the latest dev branch, and display the breaking change report to the github page.
image

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/ci/util.py [hunk 1]
```diff
@@ -117,3 +123,43 @@ def get_index_data():
...
+            elif start == '$(System.PullRequest.TargetBranch)':
+                # default value if ADO_PULL_REQUEST_TARGET_BRANCH not set in ADO
+                continue
+            else:
+                cmd = cmd_tpl.format(start=start, end=end,
+                                     code_dir=src_d_full)
+                if not check_output(shlex.split(cmd)):
+                    continue
+
+        diif_ref.append((pkg_name, src_d_full))
```
Comment (kairu-ms) [inline scripts/ci/util.py:160]: diff_ref?

**Review by kairu-ms — APPROVED** [review]

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Need support

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Need support

**Pull request merged**